javascript/ql/src/semmle/javascript/security/dataflow/ClientSideUrlRedirectCustomizations.qll

```diff
@@ -166,4 +166,22 @@ module ClientSideUrlRedirect {
       )
     }
   }
+
+  /**
+   * A write to an React attribute which may execute JavaScript code.
+   */
+  class ReactAttributeWriteUrlSink extends ScriptUrlSink {
+    ReactAttributeWriteUrlSink() {
+      exists(JSXAttribute attr | attr.getName() = propertyNameIsInterpretedAsJavaScriptUrl() |
+        this = attr.getValue().flow()
```

Comment:
Could you also check that the JSX element has a lower-case name so we're actually creating an HTML element and we're not just passing a props named `href` to another React component?

I wish there was already a predicate to check that in `JSXElement` - feel free to add one.